recipes/libpng/all/conanfile.py

```diff
@@ -131,59 +127,66 @@ def _libpng_cmake_system_processor(self):
             return "powerpc"
         return str(self.settings.arch)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["PNG_TESTS"] = False
-        self._cmake.definitions["PNG_SHARED"] = self.options.shared
-        self._cmake.definitions["PNG_STATIC"] = not self.options.shared
-        self._cmake.definitions["PNG_DEBUG"] = self.settings.build_type == "Debug"
-        self._cmake.definitions["PNG_PREFIX"] = self.options.api_prefix
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PNG_TESTS"] = False
+        tc.variables["PNG_SHARED"] = self.options.shared
+        tc.variables["PNG_STATIC"] = not self.options.shared
+        tc.variables["PNG_DEBUG"] = self.settings.build_type == "Debug"
+        tc.variables["PNG_PREFIX"] = self.options.api_prefix
         if cross_building(self):
-            self._cmake.definitions["CONAN_LIBPNG_SYSTEM_PROCESSOR"] = self._libpng_cmake_system_processor
+            system_processor = self.conf_info.get("tools.cmake.cmaketoolchain:system_processor",
+                                                  self._libpng_cmake_system_processor, check_type=str)
+            tc.cache_variables["CMAKE_SYSTEM_PROCESSOR"] = system_processor
```

Comment:
Interesting. What happen if a consumer set CMAKE_SYSTEM_PROCESSOR from a custom toolchain instead of `tools.cmake.cmaketoolchain:system_processor`?

Reply:
Good question 🤔

Reply:
Anyway, I think it will be removed in future, as CMakeToolchain can be configured directly from global.conf. So people will need to adapt in their profiles.